server/models/db/appointment.js

```diff
@@ -47,8 +47,40 @@ async function cancelWorkerAppointments(workerId, specificDate) {
     return db.any(`update ${Tables.appointment} set status='${AppointmentStatus.cancelled}' from ${Tables.appointment} a1 inner join ${Tables.workerTimeslot} w1 on a1.worker_timeslot_id = w1.worker_timeslot_id ${queryCondition};`);
 }
 
+
+// This method updates all the worker's appointments to cancelled on a specific date.
+// Note that if the worker does not have any appointments on this date, no rows will be updated (and no error will be thrown).
+
+// Cancels specific appointments/meeting and updates worker availability to available for a worker for the specific timeslot. 
+// Note that appointments/meetings are synonymous, but only appointments will be used in the backend to maintain consistency.
+async function cancelSpecificAppointment(appointmentId) {
+    //how to know if update was successful? 
+    const queryCondition = `where appointment_id='${appointmentId}'`;
+    // Isolated the query condition to decrease the length of the query line and increase code readability.
+    return await db.any(`update ${Tables.appointment} set status='${AppointmentStatus.cancelled}' ${queryCondition};`);
+}
+
+async function appointmentExists(appointmentId) {
+    //how to know if update was successful? 
+    // Isolated the query condition to decrease the length of the query line and increase code readability.
+    const queryCondition = `where appointment_id='${appointmentId}'`;
+    const queryOutput = await db.any(`select exists(select 1 from ${Tables.appointment} ${queryCondition});`);
```

Comment:
Hi Peter,
An alternative suggestion may be to select from the appointment table where the appointment id matches and check the size of the returning payload. If it is empty or null, then that would indicate that the appointment does not exist in the appointment table.

Reply:
Amy, I had initially tried the solution you mentioned and it wasn't working for me. Perhaps I simply incorrectly structured my query, but I'm having issues meaningfully actioning my queries. This piece of code I found online made the fix easy.

Reply:
Hey Amy, upon further inspection I have been able to use a simple array output of a query to reduce the total number of queries. I then use lodash '_' .map to pull the necessary information out of the array. This has enabled me to implement my goals with all the helper methods I had and you commented on.